runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp

```diff
@@ -199,10 +203,282 @@ void J9::ARM64::JNILinkage::restoreJNICallOutFrame(TR::Node *callNode, bool tear
+   if (passThread)
+      {
+      numIntegerArgs += 1;
+      }
+   // if fastJNI do not pass the receiver just evaluate the first child
+   if (!passReceiver)
+      {
+      TR::Node *firstArgChild = callNode->getChild(firstArgumentChild);
+      cg()->evaluate(firstArgChild);
+      cg()->decReferenceCount(firstArgChild);
+      firstArgumentChild += 1;
+      }
+   /* Step 1 - figure out how many arguments are going to be spilled to memory i.e. not in registers */
+   for (i = firstArgumentChild; i < callNode->getNumChildren(); i++)
+      {
+      child = callNode->getChild(i);
+      childType = child->getDataType();
+
+      switch (childType)
+         {
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+         case TR::Int64:
+         case TR::Address:
+            if (numIntegerArgs >= properties.getNumIntArgRegs())
+               numMemArgs++;
+            numIntegerArgs++;
+            break;
+
+         case TR::Float:
+         case TR::Double:
+            if (numFloatArgs >= properties.getNumFloatArgRegs())
+                  numMemArgs++;
+            numFloatArgs++;
+            break;
+
+         default:
+            TR_ASSERT(false, "Argument type %s is not supported\n", childType.toString());
+         }
+      }
+
+   // From here, down, any new stack allocations will expire / die when the function returns
+   TR::StackMemoryRegion stackMemoryRegion(*trMemory());
+   /* End result of Step 1 - determined number of memory arguments! */
+   if (numMemArgs > 0)
+      {
+      pushToMemory = new (trStackMemory()) TR::ARM64MemoryArgument[numMemArgs];
+
+      argMemReg = cg()->allocateRegister();
+      }
+
+   totalSize = numMemArgs * 8;
+   // align to 16-byte boundary
+   totalSize = (totalSize + 15) & (~15);
+
+   numIntegerArgs = 0;
+   numFloatArgs = 0;
+
+   if (passThread)
+      {
+      // first argument is JNIenv
+      numIntegerArgs += 1;
+      }
+
+   for (i = firstArgumentChild; i < callNode->getNumChildren(); i++)
+      {
+      TR::MemoryReference *mref = NULL;
+      TR::Register *argRegister;
+      TR::InstOpCode::Mnemonic op;
+      bool           checkSplit = true;
+
+      child = callNode->getChild(i);
+      childType = child->getDataType();
+
+      switch (childType)
+         {
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+         case TR::Int64:
+         case TR::Address:
+            if (childType == TR::Address)
+               {
+               argRegister = pushJNIReferenceArg(child);
+               checkSplit = false;
+               }
+            else if (childType == TR::Int64)
+               argRegister = pushLongArg(child);
+            else
+               argRegister = pushIntegerWordArg(child);
+
+            if (numIntegerArgs < properties.getNumIntArgRegs())
+               {
+               if (checkSplit && !cg()->canClobberNodesRegister(child, 0))
```

Comment:
What is this check with `checkSplit` for?

Reply:
If `checkSplit` is true, `canClobberNodesRegister` is checked. If node's register cannot be overwritten, we allocate another register and copy the value of node's register to it.
As `pushJNIReferenceArg` does the similar thing, `checkSplit` is set to false at line 311.
I borrowed this code (and variable name) from PPCJNILinkage.

Reply:
Now I think I understand what the check is for, including the similar check in #8001.